px4_ros2_cpp/src/control/vtol.cpp

```diff
@@ -118,5 +133,61 @@ void VTOL::to_fixedwing()
     RCLCPP_WARN(_node.get_logger(), "Current VTOL vehicle state unknown. Not able to transition.");
   }
 }
+
+Eigen::Vector3f VTOL::compute_acceleration_setpoint_during_transition()
+{
+  const Eigen::Vector2f velocity_xy_direction = {std::cos(_vehicle_heading), std::sin(_vehicle_heading)}; 
+  bool is_backtransition = (VTOL::get_current_state() == VTOL::State::TRANSITION_TO_MULTICOPTER) ? true : false; 
+
+  float pitch_setpoint = 0.f; 
+
+  // if we are doing a backtransition and we have gps 
+  if(is_backtransition && _vehicle_position_xy_valid && _vehicle_velocity_xy_valid){
+    pitch_setpoint = VTOL::compute_pitch_setpoint_during_backtransition(); 
+  }
+
+  Eigen::Vector2f acceleration_setpoint_during_transition = tanf(pitch_setpoint) * CONSTANTS_ONE_G * -velocity_xy_direction;
+
+  return {acceleration_setpoint_during_transition.x(), acceleration_setpoint_during_transition.y(), NAN};
+
 }
-// namespace px4_ros2
+
+float VTOL::compute_pitch_setpoint_during_backtransition(){
+
+  float deceleration_setpoint = VT_B_DEC_MSS; 
+
+  const Eigen::Vector2f velocity_xy_direction = {std::cos(_vehicle_heading), std::sin(_vehicle_heading)}; 
+  const Eigen::Vector2f back_transition_end_pos = _backtransition_commanded_position + BACK_TRANS_END_DIST * velocity_xy_direction; 
+  const float dist_to_end_pos_in_moving_direction = back_transition_end_pos.dot(velocity_xy_direction); 
+
+  if (dist_to_end_pos_in_moving_direction > __FLT_EPSILON__) {
+    deceleration_setpoint = _vehicle_velocity_xy.squaredNorm() / (2.f * dist_to_end_pos_in_moving_direction);
```

Comment:
@mahima-yoga I would not bother doing this, but would just use a default deceleration setpoint and add a method to update it.